File: rotate-linked-list/RotateLinkedList.java
public class RotateLinkedList {
    static class Node {
        public int value;
        public Node next;
        public Node(int value) { this.value = value; next = null; }
    }

    public Node RotateLinkedLists(Node head, int k) {
        int index = 0;
        while(index <= k) {
            Node current = head;
            Node tail = null;
            while(current != null) {
                tail = current;
                current = current.next;
            }
            Node next = head.next; head.next = null;
            tail.next = head; head = next;
            index++;
        }
        return head;
    }
}

File: rotate-linked-list/RotateLinkedList.swift
import Foundation 

public class RotateLinkedList {
    class Node {
        var value: Int
        var next: Node?
        init(value: Int) {
            self.value = value
        }
    }

    func Node rotateLinkedList(_ head: Node?, _ k: Int) -> Node {
        var linkedListHead = head
        var index = 0
         while index <= k {
            var current: Node? = linkedListHead
            var tail: Node? = nil
            while(current != nil) {
                tail = current
                current = current!.next
            }
            var next = linkedListHead!.next; linkedListHead!.next = nil
            tail!.next = linkedListHead; linkedListHead = next!;
            index += 1
         }
         return linkedListHead!
    }
}